src/mapml/layers/MapLayer.js

```diff
@@ -63,6 +63,16 @@ export var MapMLLayer = L.Layer.extend({
             this._container.style.zIndex = this.options.zIndex;
         }
     },
+    // remove all the extents before removing the layer from the map
+    _removeExtents: function(map){
```

Comment:
Looks like the MapMLLayer._extents._queries property should be purged here, too.